HadesAPI/router.go

```diff
@@ -45,7 +46,13 @@ func AddBuildToQueue(c *gin.Context) {
 
 	task := asynq.NewTask(payload.Name, json_payload)
 	queuePriority := utils.PrioFromInt(payload.Priority)
-	info, err := AsynqClient.Enqueue(task, asynq.Queue(queuePriority))
+	info, err := AsynqClient.Enqueue(
+		task,
+		asynq.Queue(queuePriority),
+		asynq.Retention(time.Duration(cfg.RetentionTime)*time.Minute), // Keep the result for 30 minutes
+		asynq.Timeout(0),  // No timeout
+		asynq.MaxRetry(3), // Retry 3 times
```

Comment:
I think these two should also be configurable via the environment

Reply:
Done ✅